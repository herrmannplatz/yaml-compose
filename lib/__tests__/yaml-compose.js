/* eslint-env jest */
const path = require('path')
const { build } = require('../yaml-compose')

const fixturePath = path.join(__dirname, 'fixtures')

describe('build', () => {
  it('should throw error if source is missing', () => {
    expect(() => build()).toThrowError()
  })

  it('should handle files without includes', () => {
    const singleFilePath = path.join(fixturePath, 'no-include', 'source.yml')
    expect(build(singleFilePath)).toMatchSnapshot()
  })

  it('should include single file', () => {
    const singleFilePath = path.join(fixturePath, 'single-include', 'source.yml')
    expect(build(singleFilePath)).toMatchSnapshot()
  })

  it('should include multiple file', () => {
    const singleFilePath = path.join(fixturePath, 'array-include', 'source.yml')
    expect(build(singleFilePath)).toMatchSnapshot()
  })
})
