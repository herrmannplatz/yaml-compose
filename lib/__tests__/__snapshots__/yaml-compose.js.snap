// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build should handle files without includes 1`] = `
"image: alpine:latest

variables:
  POSTGRES_USER: root
  POSTGRES_PASSWORD: secure_password

stages:
  - build
  - test
  - production

production:
  environment:
    url: https://domain.com
"
`;

exports[`build should include multiple file 1`] = `
"image: 'alpine:latest'
variables:
  POSTGRES_USER: user
  POSTGRES_PASSWORD: testing_password
  POSTGRES_DB: $CI_ENVIRONMENT_SLUG
stages:
  - build
  - test
  - production
production:
  environment:
    url: 'https://$CI_PROJECT_PATH_SLUG.$AUTO_DEVOPS_DOMAIN'
    name: production
  stage: production
  script:
    - install_dependencies
    - deploy
  only:
    - master
"
`;

exports[`build should include single file 1`] = `
"image: 'alpine:latest'
variables:
  POSTGRES_USER: user
  POSTGRES_PASSWORD: testing_password
  POSTGRES_DB: $CI_ENVIRONMENT_SLUG
stages:
  - build
  - test
  - production
production:
  environment:
    url: 'https://$CI_PROJECT_PATH_SLUG.$AUTO_DEVOPS_DOMAIN'
    name: production
  stage: production
  script:
    - install_dependencies
    - deploy
  only:
    - master
"
`;
